tests/test_redis_pytest.py

```diff
@@ -81,6 +97,46 @@ def test_run_should_run():
     assert result.ret == EXIT_OK
 
 
+def test_elborate_test_modules(testdir, redis_connection, redis_args):
+    """Ensure that modules in are collected and executed correctly."""
+    folders = {'module1': ["file1.py"],
+               'module2': ["file2.py"],
+               'module3': ["file3.py"]}
+    text_mapping = {}
+    num_test_methods_per_file = 2
+    num_times_to_run_tests = 2
+    test_to_run_list = []
+    # Create the contents of the test files
+    for folder in folders.keys():
```

Comment:
hard to read whats going on here because the setup is part of the test, i think it would benefit from extracting it into a fixture or make this setup part of the create_elaborate_test_directories method.